tests/integration/kubernetes/tests_common.sh

```diff
@@ -138,14 +152,18 @@ create_common_genpolicy_settings() {
 
 	auto_generate_policy_enabled || return 0
 
+	adapt_common_policy_settings "${default_genpolicy_settings_dir}"
+
 	cp "${default_genpolicy_settings_dir}/genpolicy-settings.json" "${genpolicy_settings_dir}"
 	cp "${default_genpolicy_settings_dir}/rules.rego" "${genpolicy_settings_dir}"
 
 	# Set the default namespace of Kata CI tests in the genpolicy settings.
 	set_namespace_to_policy_settings "${genpolicy_settings_dir}" "${TEST_CLUSTER_NAMESPACE}"
 
-	# allow genpolicy to access containerd without sudo
-	sudo chmod a+rw /var/run/containerd/containerd.sock
+	if [ "${GENPOLICY_PULL_METHOD}" == "containerd" ]; then
+		# allow genpolicy to access containerd without sudo
+		sudo chmod a+rw /var/run/containerd/containerd.sock
```

Comment:
So, for TDX this won't fly, as we're using k3s as the way to deploy k8s.

You should look for the `KUBERNETES` env var, if it's k3s, the containerd socket path is  `/run/k3s/containerd/containerd.sock`